src/main/java/it/reply/orchestrator/service/VaultServiceImpl.java

```diff
@@ -0,0 +1,163 @@
+    this.vaultProperties = vaultProperties;
+    this.token = "";
+  }
+
+  private VaultTemplate getTemplate() {
+    return new VaultTemplate(VaultEndpoint.create(
+        vaultProperties.getUrl(),
+        vaultProperties.getPort()),
+        new TokenAuthentication(token));       
+  }
+
+  public VaultResponse writeSecret(String path, Object secret) {
+    return getTemplate().write(path, secret);
+  }
+
+  public <T> T readSecret(String path, Class<T> type) {        
+    return getTemplate().read(path, type).getData();
+  }
+
+  public Map<String,Object> readSecret(String path) {
+    return getTemplate().read(path).getData();
+  }
+
+  public void deleteSecret(String path) {
+    getTemplate().delete(path);
+  }
+
+  public List<String> listSecrets(String path) {
+    return getTemplate().list(path);
+  }
+
+  /**
+   * Retrieve Vault token starting from access token.
+   */
+  @SuppressWarnings("unchecked")
+  public VaultService retrieveToken(String accessToken) {
+    String token = "";
+    try {
+      VaultEndpoint endpoint = VaultEndpoint.create(
+          vaultProperties.getUrl(),
+          vaultProperties.getPort());
+      URI uri = endpoint.createUri("auth/jwt/login");
+
+      CloseableHttpClient httpclient = HttpClients.createDefault();
+      HttpPost httpPost = new HttpPost(uri.toString());
+      String json = "{\"jwt\":\"" + accessToken + "\"}";
+      HttpEntity stringEntity = new StringEntity(json,ContentType.APPLICATION_JSON);
+      httpPost.setEntity(stringEntity);
+      CloseableHttpResponse response = httpclient.execute(httpPost);            
+
+      StringBuilder responseStrBuilder = new StringBuilder();
+
+      BufferedReader reader = new BufferedReader(
+          new InputStreamReader((response.getEntity().getContent())));
+
+      String inputStr;
+      while ((inputStr = reader.readLine()) != null) {
+        responseStrBuilder.append(inputStr);
+      }
+      
+      final int status = response.getStatusLine().getStatusCode();
+
+      reader.close();
+      response.close();
+      httpclient.close();
+
+      if (status == 200) {
+
+        HashMap<String,Object> result =
+            new ObjectMapper().readValue(responseStrBuilder.toString(), HashMap.class);         
+
+        HashMap<String,Object> auth = (HashMap<String,Object>) result.get("auth");
+
+
+        token = (String) auth.get("client_token");
+      } else {
+        String message = responseStrBuilder.toString();
+        if (status == 400) {
+          if (message.toLowerCase().contains("token is expired")) {
+            throw new VaultTokenExpiredException("Unable to retrieve token for Vault:"
+                + " accessToken is expired");
+          }
+        }
+        throw new VaultException(String.format("Unable to retrieve token for Vault:"
+            + " %d (%s).", status, message));
+      }        
+    } catch (VaultException ve) {            
+      throw ve;
+    } catch (Exception e) {
```

Comment:
Please do not catch generic `Exception`s. It with catch also eventual `InterruptedException`s without proper handling